tests/e2e/support/page/publicLinkPage.ts

```diff
@@ -0,0 +1,55 @@
+import { BrowserContext, Page } from 'playwright'
+import { state } from '../../cucumber/environment/shared'
+import { lastCreatedPublicLink } from './files/publicLink'
+import { expect } from '@playwright/test'
+import util = require('util')
+
+export class PublicLinkPage {
+  public context: BrowserContext
+  private static page: Page
+  private readonly passwordInput: string
+  private readonly passwordSubmitButton: string
+  private readonly fileUploadInput: string
+  private uploadedFiles: string
+
+  constructor() {
+    this.passwordInput = 'input[type="password"]'
+    this.passwordSubmitButton =
+      '//*[@id="password-submit"]|//*[@id="oc-textinput-3"]/ancestor::div[contains(@class, "oc-mb-s")]/following-sibling::button'
+    this.uploadedFiles = `text='%s'`
+    this.fileUploadInput = '//input[@id="file_upload_start" or @class="dz-hidden-input"]'
+  }
+
+  async setup(): Promise<void> {
+    this.context = await state.browser.newContext({ ignoreHTTPSErrors: true })
+    PublicLinkPage.page = await this.context.newPage()
+  }
```

Comment:
what is the purpose of this setup function?
Put these inside the constructor. I think that is what you want.

Reply:
in typescript, we can't use async inside the constructor